collect.py

```diff
@@ -222,10 +226,120 @@ def get_domtree(driver, file_path):
         f.write(driver.execute_script('return document.documentElement.outerHTML'))
 
 
+def get_coordinates(driver, file_path):
+    dom_tree = etree.HTML(driver.execute_script('return document.documentElement.outerHTML'))
+    dom_element_tree = etree.ElementTree(dom_tree)
+    loc_dict = {}
+    dom_tree_elements = [elem for elem in dom_tree.iter(tag=etree.Element)]
+    web_elements = driver.find_elements_by_css_selector('*')
+    left_elements = []
+    dom_xpaths = []
+
+    for element in dom_tree_elements:
+        xpath = dom_element_tree.getpath(element)
+        dom_xpaths.append(xpath)
+    for element in web_elements:
+        xpath = driver.execute_script("""function absoluteXPath(element) {
```

Comment:
Since this is a huge script, I think we should write it in a separate JS file and load this JS file at the startup of the crawler in a variable, and then use the variable here.

Reply:
Did you find this implementation online or did you implement it from scratch?

Reply:
I found the script online to which I had to make some changes so that the xpath generated by the script and that we find using lxml come out to be exactly the same. Otherwise they both won't be compatible for future part of xpert.

Reply:
OK! Where did you find it? We should make sure the license is compatible, or if you found it on stackoverflow you can add a comment at the beginning of the file with a link to the stackoverflow answer.

Reply:
https://stackoverflow.com/questions/47069382/want-to-retrieve-xpath-of-given-webelement

I will add it as a comment in the beginning of the file.